core/src/main/scala/chisel3/internal/firrtl/Converter.scala

```diff
@@ -93,6 +93,9 @@ private[chisel3] object Converter {
     // TODO Simplify
     case lit: ILit =>
       throw new InternalErrorException(s"Unexpected ILit: $lit")
+    case PropertyLit(lit: Int) => fir.IntegerPropertyLiteral(lit)
+    case PropertyLit(lit: Long) => fir.IntegerPropertyLiteral(lit)
+    case PropertyLit(lit: BigInt) => fir.IntegerPropertyLiteral(lit)
```

Comment:
It might be better to do something more typeclassy here like we did for Property types.... I am taking a look at how to do something like that here.

Reply:
I tried this out and pushed it up in https://github.com/chipsalliance/chisel/pull/3482/commits/03d2034d1afbf33f91995acedf9a3d7860cfebc3, where now the PropertyType typeclass is responsible for handing out literals, like how it hands out types. Curious if this seems better. What do you think @seldridge @jackkoenig @mwachs5? I'm happy to go either route, and I see Megan approved the initial version.